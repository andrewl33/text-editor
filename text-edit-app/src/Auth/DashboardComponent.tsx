import * as React from "react";
import { ListComponent } from "../generic/Lists/ListComponent";
import { DashboardComponentProps } from "../types";

export const DashboardComponent = (
  props: DashboardComponentProps
): React.ReactElement<DashboardComponentProps> => {
  return (
    <div>
      {props.dashboard!.collections && props.onCollectionClick && (
        <ListComponent
          header="Collections"
          items={...props.dashboard!.collections}
          onClickToPage={props.onCollectionClick}
        />
      )}
      {props.dashboard!.files && (
        <ListComponent
          header="Files"
          items={...props.dashboard!.files}
          onClickToPage={props.onFileClick}
        />
      )}
    </div>
  );
};
