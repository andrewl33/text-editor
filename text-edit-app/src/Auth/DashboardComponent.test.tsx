import * as React from "react";
import * as renderer from "react-test-renderer";
import { DashboardComponent } from "./DashboardComponent";

const authProps = {
  dashboard: { collections: [], files: [] }
};

const dispatchProps = {
  onCollectionClick: jest.fn(),
  onFileClick: jest.fn()
};

const props = {
  ...authProps,
  ...dispatchProps
};

describe("DashboardComponent", () => {
  test("renders correctly", () => {
    const component = renderer.create(<DashboardComponent {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
