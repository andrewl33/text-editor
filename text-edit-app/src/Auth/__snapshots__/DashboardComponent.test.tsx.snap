// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent renders correctly 1`] = `
<div>
  <div>
    <table
      className="ui blue large celled inverted middle aligned table"
      divided="true"
      relaxed="true"
      style={
        Object {
          "borderRadius": 0,
          "color": "rgba(255,255,255,.9)",
        }
      }
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
            colSpan="2"
          >
            Collections
          </th>
          <th
            className=""
            colSpan="2"
          >
            Dates
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      />
    </table>
  </div>
  <div>
    <table
      className="ui blue large celled inverted middle aligned table"
      divided="true"
      relaxed="true"
      style={
        Object {
          "borderRadius": 0,
          "color": "rgba(255,255,255,.9)",
        }
      }
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
            colSpan="2"
          >
            Files
          </th>
          <th
            className=""
          >
            Tags
          </th>
          <th
            className=""
            colSpan="2"
          >
            Dates
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      />
    </table>
  </div>
</div>
`;
